woocommerce/woocommerce-android
# Shipping rates custom weight support

Part of: #12286

### Description
This PR supports custom weight in shipping rates by adding the custom view input field and refactoring the main flow's view model to observe and respond to typing events. We are no handling errors in this task, we will work on errors on M2.

### Testing information

1. Open the orders section
2. Select an order that is eligible to create shipping labels (processing state)
3. Tap on create shipping label
4. Select a package
5. Check that shipping rates are fetched from the API
6. Check that the custom weight field is displayed
7. Check that the custom weight hint matches the sum of the package weight and items weight
8. Type a new weight in the custom weight input field
9. Check that a new request is made to refresh the shipping rates

### The tests that have been performed
1. Checked that changing the custom weight refreshes the shipping rates
2. Checked that deleting a custom weight refreshes the shipping rates
3. Checked that the custom weight hint matches  the sum of package weight and items weight
4. Checked that after changing the custom weight, the UI waits for the user to stop typing before requesting the data to the API

### Images/gif

https://github.com/user-attachments/assets/f8aabc9d-364e-47be-a4fb-6e969225ee60


## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:
